optuna/storages/_base.py

```diff
@@ -556,7 +555,11 @@ def get_n_trials(self, study_id: int, state: Optional[TrialState] = None) -> int
             :exc:`KeyError`:
                 If no study with the matching ``study_id`` exists.
         """
-        raise NotImplementedError
+        self.read_trials_from_remote_storage(study_id)
```

Comment:
Sorry for the delayed comment. `read_trials_from_remote_storage` is actually meant to be called from outside the storage, to sync only when necessary, and currently only from the `_CachedStorage`. Given that this `_CachesStorage` is private and not expected to have been used directly by the user, I think we can safely remove this line. 
```suggestion
```